main.cpp

```diff
@@ -13,75 +13,506 @@ What you create in this project will be used as the basis of Project 5 in the co
+    3) the min amount of fuel it can hold before it starts to effect performance // double
+    4) capacity sensors // int
+    5) material // string
+3 things it can do:
+    1) notify the rider when it is low on fuel
+    2) notify the rider when the wrong type of fuel has been used
+    3) hold fuel
+ */
+
+struct FuelTank
+{
+    // capacity in litres
+    double capacityL = 8.5;
+    // the type of fuel it can hold
+    std::string fuel = "unleaded";
+    // the min amount of fuel it can hold before it starts to effect performance
+    double recommendedMinFuel = 3.5;
+    // capacity sensors
+    int sensors = 4;
+    // material
+    std::string material = "steel";
+
+    // notify the rider when it is low on fuel
+    std::string lowFuel(double recommendedMinFuel = 3.5, bool fuelIsLowerThanRecommendedAmoutn = true); // return type as std::string is ok?
+    // notify the rider when the wrong type of fuel has been used
+    std::string wrongFuel(int sensors = 4, std::string fuel = "unleaded", bool wrongFuelInTank = true);
+    // hold fuel
+    void holdFuel(double capacityL = 8.5, double recommendedMinFuel = 3.5);
+};
+
+/*
+Thing 9) Tyre
+5 properties:
+    1) type of material // string
+    2) thickness of the material // double
+    3) average lifespan in km travelled  // double
+    4) the manufacturer // string
+    5) the type of terrain the tyre is most suited for // string
+3 things it can do:
+    1) puncture
+    2) provide support for the bike
+    3) wear out over time
+ */
+
+struct Tyre
+{
+    // type of material // string
+    std::string material;
+    // hickness of the material // double
+    double thickness;
+    // average lifespan in km travelled  // double
+    double lifespanKM = 2500;
+    // the manufacturer // string
+    double pressure = 14;
+    // the type of terrain the tyre is most suited for // string
+    std::string terrain;    
+
+    // puncture
+    void puncture(bool tyrePressureDroppedBelow5 = true);
+    // provide support for the bike
+    void supportVehicle(std::string vehicleType = "Motor Bike", bool numOfTyres = 2);
+    // wear out over time
+    void wear(double lifespanKM);
+};
+
+ /*
+Thing 10) Motorbike
+5 properties:
+    1) seat 
+    2) engine 
+    3) exhaust 
+    4) fuel tank 
+    5) tyre
+3 things it can do:
+    1) Move forwards when the throttle is pressed
+    2) Stand stationary using its stand
+    3) Hold fuel in the fuel tank
+ */
+ 
+struct Motorbike
+{
+    // seat
+    Seat seat;
+    // engine 
+    Engine engine;
+    // exhaust 
+    Exhaust exhaust;
+    // fuel tank 
+    FuelTank fuelTank;
+    // tyre 
+    Tyre tyre;
+
+    // Move forwards when the throttle is pressed
+    void move(std::string direction, int speed);
+    // Stand stationary using its stand
+    void stand(bool useStand = true, bool tyresMove = false);
+    // Hold fuel in the fuel tank
+    void holdFuel(FuelTank fuelTank, bool activateFuelSensors = true);
```

Comment:
The class already has a fuel tank as a member variable.
Why are you passing one as a function parameter?
- [ ] revise